develop/DSI/DSIOpenConfig/DSI_OpenConfig_Middleware.md

```diff
@@ -0,0 +1,456 @@
+---
+uid: DSI_OpenConfig_Middleware
+---
+# OpenConfig Middleware
+
+## Purpose
+
+The purpose of the OpenConfig middleware is to make it possible to **easily consume data exposed over the gNMI service in a DataMiner connector**. It assists you in **setting up the connection with the endpoint**, provides **easy to use wrappers around the gRPC service calls**, and makes it possible to **map specific paths in the YANG model to DataMiner parameters** so the notifications the service sends out get automatically parsed and the parameters are populated with data.
+
+## Prerequisites
+
+In order to use the OpenConfig middleware, you will need to have the following setup:
+
+- A DataMiner Agent that is [connected to dataminer.services](xref:Connecting_your_DataMiner_System_to_the_cloud) and runs DataMiner 10.3.3 or higher.
+
+- The [CommunicationGateway DxM](xref:DataMinerExtensionModules#communicationgateway) is installed on at least one of the DataMiner Agents in the cluster.
+
+> [!NOTE]
+> You can install the CommunicationGateway DxM with the Admin app. See [Managing the nodes of a DMS connected to dataminer.services](xref:Managing_cloud-connected_nodes).
+
+## Usage
+
+### Including the middleware package
+
+Include the NuGet package *Skyline.DataMiner.DataSources.OpenConfig.Gnmi* in your project, as explained under [Consuming NuGet packages](xref:Consuming_NuGet).
+
+When you are developing a connector, there is an optional package when you want to use the [DataMapper](#datamapper): *Skyline.DataMiner.DataSources.OpenConfig.Gnmi.Protocol*
+
+### Setting up the connection with the endpoint
+
+#### Creating a GnmiClient
```

Comment:
Adding more structure to the text underneath this header would make it more user-friendly. I recommend dividing this into subsections, for example one for each item that has to be passed.

Reply:
Done